packages/hs-react-ui/src/components/Button/ButtonContainers.tsx

```diff
@@ -1,44 +1,86 @@
-import styled from 'styled-components';
-import { darken } from 'polished';
-import { ButtonTypes, ButtonContainer } from './Button';
+import styled, { StyledComponentBase } from 'styled-components';
+import { darken, readableColor } from 'polished';
 import colors from '../../enums/colors';
+import fonts from '../../enums/fonts';
 
-export const PrimaryButtonContainer = styled(ButtonContainer)`
-  color: ${colors.background};
-  background-color: ${colors.primary};
-  &:hover {
-    background-color: ${darken(0.05, colors.primary)};
-  }
-  &:active {
-    background-color: ${darken(0.1, colors.primary)};
-  }
-`;
+export type ButtonContainerProps = {
+  elevation?: number;
+  color?: string;
+  type: string;
+};
+export enum ButtonTypes {
+  primary = 'primary',
+  secondary = 'secondary',
+  default = 'default',
+  destructive = 'destructive',
+  link = 'link',
+  outline = 'outline',
```

Comment:
Maybe you could have it so that if type isn't in the list of normal types - you can try to see if they're trying to access a color in the color constants object? Otherwise default to fill I guess or throw angry error messages

Reply:
For me, `type` should more about how the color they specified is applied, and the general shape of the button. Using type to control color when we already expose a color prop will make it confusing as to which has priority

Reply:
I went ahead and removed primary/secondary/default/destructive and replace it with fill. Without type provided, the button defaults to fill. Without color provided, the button defaults to the old 'default' button that is lightgray background with dark gray text